src/tests/end-to-end/tests/details-view/tabstop.test.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { launchBrowser } from '../../common/browser-factory';
+import { fastPassSelectors } from '../../common/element-identifiers/common-selectors';
+import { popupPageElementIdentifiers } from '../../common/element-identifiers/popup-page-element-identifiers';
+import { Page } from '../../common/page-controllers/page';
+import { TargetPage } from '../../common/page-controllers/target-page';
+import { Browser } from './../../common/browser';
+import { DEFAULT_PAGE_ELEMENT_WAIT_TIMEOUT_MS } from './../../common/timeouts';
+
+describe('Tabstop tests', () => {
+    describe('tabstop from fastpass', () => {
+        let browser: Browser;
+        let targetPage: TargetPage;
+        let fastPassPage: Page;
+
+        beforeAll(async () => {
+            browser = await launchBrowser({ suppressFirstTimeDialog: true });
+            targetPage = await browser.newTargetPage();
+            fastPassPage = await gotoFastPass(browser, targetPage.tabId);
+        });
+
+        afterAll(async () => {
+            if (browser) {
+                await browser.close();
+            }
+        });
+
+        test('if visualHelper is turned on after starting tabstop and pressing tabs on target page', async () => {
+            const toggleAriaLabel = 'Tab stops';
+
+            await goToTabStopTest();
+            await enableToggleByAriaLabel(toggleAriaLabel);
+
+            await targetPage.bringToFront();
+
+            // press tabs 3 times. todo: DRY this code
+            await targetPage.keyPress('Tab');
+            await targetPage.keyPress('Tab');
+            await targetPage.keyPress('Tab');
+
+            expect(await targetPage.getShadowRootHtmlSnapshot()).toMatchSnapshot();
+        });
+
+        async function gotoFastPass(browserLocal: Browser, targetTabId: number): Promise<Page> {
+            const popupPage = await browserLocal.newPopupPage(targetPage);
+            let fastPass: Page;
+            await Promise.all([
+                browserLocal.waitForDetailsViewPage(targetPage).then(page => (fastPass = page)),
```

Comment:
do we need to use the details view page??? why can we not enable tab stops from the popup page, ad hoc tools view?

Reply:
I wanted to create one test with the flow from details page and then add similar test cases for other scenarios like popup-> adhoc tools view; which will be in the followup PRs on this